pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -522,13 +508,15 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 
 	// Check for any conditions that might require scheduling retry
 
-	// When pod is removed from scheduling queue because of deletion or any
-	// other reasons, binding operation should be cancelled. There is no need
-	// to check PV/PVC bindings any more.
-	// We check pod binding cache here which will be cleared when pod is
-	// removed from scheduling queue.
-	if b.podBindingCache.GetDecisions(pod) == nil {
-		return false, fmt.Errorf("pod %q does not exist any more", podName)
+	// When pod is deleted, binding operation should be cancelled. There is no
+	// need to check PV/PVC bindings any more.
+	_, err = b.podLister.Pods(pod.Namespace).Get(pod.Name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, fmt.Errorf("pod %q does not exist any more", podName)
+		} else {
+			klog.Errorf("failed to get pod %s/%s from the lister: %v", pod.Namespace, pod.Name, err)
+		}
```

Comment:
the pod is removed from scheduling queue if 1) deleted 2) scheduled, so we can check if the pod is deleted here